### PR TITLE
Add OrderReference and ExpenseReference filters to TransactionsCollectionQuery

### PR DESCRIPTION
**Pull request by gustavlrsn**
Resolves https://github.com/opencollective/opencollective/issues/7144
Project: https://github.com/opencollective/opencollective/issues/7106

### CODE REVIEW

**Comment by gustavlrsn** [comment]
Current dependencies on/for this PR:
* main
  * **PR #9584** <a href="https://app.graphite.dev/github/pr/opencollective/opencollective-api/9584?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/opencollective/opencollective-api/9584?utm_source=stack-comment).

File: server/graphql/v2/query/collection/TransactionsCollectionQuery.ts [hunk 1]
```diff
@@ -280,9 +290,17 @@ export const TransactionsCollectionResolver = async (
...
+  if (args.expense) {
+    const expenseId = getDatabaseIdFromExpenseReference(args.expense);
+    where.push({ ExpenseId: expenseId });
+  }
   if (args.hasExpense !== undefined) {
     where.push({ ExpenseId: { [args.hasExpense ? Op.ne : Op.eq]: null } });
   }
+  if (args.order) {
+    const orderId = getDatabaseIdFromOrderReference(args.order);
+    where.push({ OrderId: orderId });
```
Comment (znarf) [inline server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:302]: What if `args.order` is bogus and `orderId` end up being `null`.

Is it gonna return all entries that have `OrderId=null` ?

Comment (gustavlrsn) [reply server/graphql/v2/query/collection/TransactionsCollectionQuery.ts:302]: @znarf it is handled by `getDatabaseIdFromOrderReference` that throws errors for invalid or missing id's. There seems to be no case to end up with returning entries that have `OrderId=null`

**Review by znarf — APPROVED** [review]
Good for me!

**Pull request merged**